src/Utility/HttpStatus.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Helper class for dealing with HTTP status codes.
+ *
+ * @package Requests\Utilities
+ */
+
+namespace WpOrg\Requests\Utility;
+
+use WpOrg\Requests\Exception\InvalidArgument;
+
+/**
+ * HTTP status codes helper class.
+ *
+ * @package Requests\Utilities
+ */
+final class HttpStatus {
+
+	const TEXT_100 = 'Continue';
+	const TEXT_101 = 'Switching Protocols';
+	const TEXT_200 = 'OK';
+	const TEXT_201 = 'Created';
+	const TEXT_202 = 'Accepted';
+	const TEXT_203 = 'Non-Authoritative Information';
+	const TEXT_204 = 'No Content';
+	const TEXT_205 = 'Reset Content';
+	const TEXT_206 = 'Partial Content';
+	const TEXT_300 = 'Multiple Choices';
+	const TEXT_301 = 'Moved Permanently';
+	const TEXT_302 = 'Found';
+	const TEXT_303 = 'See Other';
+	const TEXT_304 = 'Not Modified';
+	const TEXT_305 = 'Use Proxy';
+	const TEXT_306 = '(Unused)';
```

Comment:
The original text for this in the `WpOrg\Requests\Exception\Http\Status306` class was "'Switch Proxy'". Might be good to keep it that way ?

Looks like the `306` status code was originally for "Switch Proxy", but became unused with HTTP1.1.

I think if the code is still used/received, Requests should probably still handle it according to its original use ?

This probably also means that this constant needs a docblock this.